CMakeLists.txt

```
@@ -49,6 +49,28 @@ function(packageProject)
     endif()
 
     if(DEFINED PROJECT_VERSION_HEADER)
+      string(REGEX MATCH "^([0-9]+)(\\.([0-9]+))?(\\.([0-9]+))?(\\.([0-9]+))?$" _
```

I just realised that CMake doesn't clear previous matches when using `REGEX MATCH`, so we should unset them here to avoid bugs.

```suggestion
      # clear previous matches
      unset(CMAKE_MATCH_1)
      unset(CMAKE_MATCH_3)
      unset(CMAKE_MATCH_5)
      unset(CMAKE_MATCH_7)

      string(REGEX MATCH "^([0-9]+)(\\.([0-9]+))?(\\.([0-9]+))?(\\.([0-9]+))?$" _
```